### PR TITLE
Wip/hughsie/recoldplug fixes

### PR DESCRIPTION
**Pull request by hughsie**

### CODE REVIEW

File: src/fu-engine.c [hunk 1]
```diff
@@ -3488,7 +3491,7 @@ fu_engine_load (FuEngine *self, GError **error)
 
 	/* add devices */
 	fu_engine_plugins_setup (self);
-	fu_engine_plugins_coldplug (self);
+	fu_engine_plugins_coldplug (self, TRUE);
```
Comment (superm1) [inline src/fu-engine.c]: Hmm, why no longer allow coldplug to be called during coldplug?  Do you think that's the root of ending up with two devices?  I'll need to check and see if this works correctly for the flows I expect.

Comment (hughsie) [reply src/fu-engine.c]: The root problem was the device list was accepting two different FuDevice's with the same ID. When we recoldplug it's pointless going to all the trouble of creating a FuDevice (e.g. the amt one is slooow) just for the device-list to throw it away, so I figured opting-in to the recodplug was a good idea. The TRUE/FALSE passed to fu_engine_plugins_coldplug just signifies if the call to coldplug is "original" i.e. as part of the daemon startup, or as a recoldplug.

**Comment by hughsie** [comment]
Of course, the other way of doing this is an explicit `fu_plugin_recoldplug()` vfunc -- that might be simpler all round...

**Comment by superm1** [comment]
I prefer that approach too.

**Comment by hughsie** [comment]
Okay, patch reworked. I'm not 100% sold on the "coldplug_again" vfunc name, better ideas welcome. It's certainly a cleaner patch than the flags-based approach of before.

File: plugins/dell/fu-plugin-dell.c [hunk 2]
```diff
@@ -1114,11 +1114,23 @@ fu_plugin_startup (FuPlugin *plugin, GError **error)
 	return TRUE;
 }
 
-gboolean
-fu_plugin_coldplug (FuPlugin *plugin, GError **error)
+static gboolean
+fu_dell_coldplug (FuPlugin *plugin, GError **error)
```
Comment (superm1) [inline plugins/dell/fu-plugin-dell.c]: I think this should be `fu_plugin_dell_coldplug `per the naming scheme adopted for the others.

**Comment by superm1** [comment]
How about `fu_plugin_recoldplug` instead of `fu_plugin_coldplug_again`?  The plugins already request recoldplug, so that would make it clear what they're requesting.

**Comment by hughsie** [comment]
`fu_plugin_recoldplug` is the method they call to request a recoldplug already. We could rename that to something like `fu_plugin_request_recoldplug()`

**Comment by superm1** [comment]
Yeah that works

**Pull request merged**